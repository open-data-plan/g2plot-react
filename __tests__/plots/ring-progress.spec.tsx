import { render } from '@testing-library/react';
import React from 'react';
import RingProgressChart from '../../src/plots/ring-progress';

describe('RingProgressChart', () => {
  test('should render without crashed', () => {
    const result = render(<RingProgressChart percent={0.8} />);

    expect(result.container).toMatchSnapshot();
  });
});
