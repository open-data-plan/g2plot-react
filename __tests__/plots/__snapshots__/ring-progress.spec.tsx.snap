// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RingProgressChart should render without crashed 1`] = `
<div>
  <div
    data-chart-source-type="G2Plot"
    size-sensor-id="1"
  >
    <div
      style="position:relative;"
    >
      <canvas
        height="400"
        style="width: 400px; height: 400px; display: inline-block; vertical-align: middle;"
        width="400"
      />
      <div
        class="g2-html-annotation"
        style="position: absolute; pointer-events: none; width: 313.6px; transform: translate(-50%, -50%); overflow: hidden; white-space: nowrap; text-overflow: ellipsis; display: flex; justify-content: center; align-items: center; color: rgb(77, 77, 77); text-shadow:  0px 0px 0px; font-size: 14px; font-weight: 300; text-align: center; left: 200px; top: 200px; z-index: 7;"
      >
        80.00%
      </div>
    </div>
    <object
      class="size-sensor-object"
      data="about:blank"
      style="display: block; position: absolute; top: 0px; left: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1; opacity: 0;"
      tabindex="-1"
      type="text/html"
    />
  </div>
</div>
`;
